### PR TITLE
Add triggers to dtb

### PR DESCRIPTION
**Pull request by lachmanfrantisek**
Still WIP, submitting just to have my work somewhere.

TODO:
- [x] database schema
  - [x] finalise the table for copr builds
  - [x] add a new table for koji builds
  - [x] add a new table for TFT test runs
- [x] save data to database and load it when receiving the results
- [x] implement the migration
- [x] test migration with stage and prod content (using `pg_dump`)
- [ ] tests:
  - [x] models
  - [ ] migration (migration tests will come in the next PR)

### CODE REVIEW

File: alembic/versions/26bbdf3d6798_add_tables_for_other_triggers.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    )
+    op.add_column(
+        "copr_builds", sa.Column("job_trigger_id", sa.Integer(), nullable=True)
+    )
+    op.drop_constraint("copr_builds_pr_id_fkey1", "copr_builds", type_="foreignkey")
+    op.create_foreign_key(
+        None, "copr_builds", "build_triggers", ["job_trigger_id"], ["id"]
+    )
+    op.drop_column("copr_builds", "logs")
+    op.drop_column("copr_builds", "pr_id")
```
Comment (TomasTomecek) [inline alembic/versions/26bbdf3d6798_add_tables_for_other_triggers.py]: I know it's in your todo but I guess this is where the migration should happen

**Comment by lachmanfrantisek** [comment]
> [TomasTomecek](/TomasTomecek) changed the title ~WIP: Add triggers to dtb~ WIP: Add triggers to psql

Why did you need to change that? (I name the PRs same as the branch so it's misleading to me.)

File: packit_service/models.py [hunk 2]
```diff
@@ -77,24 +77,27 @@ def get_sa_session() -> Session:
     Base = declarative_base()
 
 
-class GitProject(Base):
+class GitProjectDtb(Base):
```
Comment (TomasTomecek) [inline packit_service/models.py]: now i get it, dtb stands for database, right?

File: packit_service/models.py [hunk 3]
```diff
@@ -122,42 +125,203 @@ class PullRequest(Base):
...
+                session.query(IssueDtb)
+                .filter_by(issue_id=issue_id, project_id=project.id)
+                .first()
+            )
+            if not pr:
+                pr = IssueDtb()
+                pr.issue_id = issue_id
+                pr.project_id = project.id
+                session.add(pr)
+            return pr
```
Comment (TomasTomecek) [inline packit_service/models.py]: s/pr/issue/g

File: packit_service/models.py [hunk 4]
```diff
@@ -122,42 +125,203 @@ class PullRequest(Base):
...
+    def __str__(self):
+        return self.__repr__()
+
+
+class ProjectReleaseDtb(Base):
+    __tablename__ = "project_releases"
+    id = Column(Integer, primary_key=True)  # our database PK
+    tag_name = Column(String)
+    project_id = Column(Integer, ForeignKey("git_projects.id"))
+    project = relationship("GitProjectDtb", back_populates="releases")
```
Comment (TomasTomecek) [inline packit_service/models.py]: would it be possible to add commit hash here?

File: packit_service/models.py [hunk 5]
```diff
@@ -122,42 +125,203 @@ class PullRequest(Base):
...
+                trigger.type = type
+                trigger.trigger_id = trigger_id
+                session.add(trigger)
+            return trigger
+
+    def get_trigger_object(self) -> AbstractTriggerDtbType:
+        with get_sa_session() as session:
+            return session.query(MODEL_FOR_TRIGGER[self.type])(
+                id=self.trigger_id
+            ).first()
```
Comment (TomasTomecek) [inline packit_service/models.py]: neat

**Comment by TomasTomecek** [comment]
@lachmanfrantisek b/c I thought it's a typo, after I read the code I realized it's not

**Comment by lachmanfrantisek** [comment]
@TomasTomecek thanks for the early review. It's far from final, but I am starting to understand (and like) sqlalchemy and alembic.

(I'll switch to the tmt/fmf for today and continue tomorrow.)

File: packit_service/models.py (hunk 2 above)
Comment (lachmanfrantisek) [reply packit_service/models.py]: Is it ok or I need to choose some better shortcut?

Comment (TomasTomecek) [reply packit_service/models.py]: honestly, this is the first time I see "Dtb" as a shortcut for database, that's why I was so confused. Usual term I was familiar with was "Db".

In django, the best practice is to suffix with "Model".

My opinion: I don't care, to me this is picking nits - my editor will complete everything anyway :) Hence I'm clicking resolve.

Comment (lachmanfrantisek) [reply packit_service/models.py]: I've moved to `Model`. I am used to it from the Django world as well...

**Comment by sakalosj** [comment]
my suggestion is to change dtb to db, if there is no specific reason to use it

**Comment by lachmanfrantisek** [comment]
> my suggestion is to change dtb to db, if there is no specific reason to use it

I've already changed it to models...;)

**Comment by sakalosj** [comment]
> > my suggestion is to change dtb to db, if there is no specific reason to use it
> 
> I've already changed it in code to models...;)

i meant like:

https://github.com/packit-service/packit-service/blob/1825e6c1c50f76975fd563a667a6e45fecddfd2e/packit_service/service/events.py#L43
https://github.com/packit-service/packit-service/blob/1825e6c1c50f76975fd563a667a6e45fecddfd2e/packit_service/service/events.py#L216
https://github.com/packit-service/packit-service/blob/1825e6c1c50f76975fd563a667a6e45fecddfd2e/packit_service/service/events.py#L263

still see it ther, but maybe changes were not yet pushed

**Comment by lachmanfrantisek** [comment]
> i meant like:

Ok, I missed that... Thanks.

**Comment by lachmanfrantisek** [comment]
recheck

**Comment by lachmanfrantisek** [comment]
Current version tested with stage and prod content and migration went well. (I would like to create some tests for it.)

**Pull request merged**